candump.c

```diff
@@ -314,9 +429,32 @@ int main(int argc, char **argv)
 			}
 			break;
 
+		case ':': //handle flags with optional values
+			
+			switch (optopt)
+			{
+			case 'l':
+				log = 1;
+
+				if (process_logname_arg(optarg) != 0) {
+					print_usage(basename(argv[0]));
+					exit(1);
+				}
+				break;
+			default:
+				fprintf(stderr, "option -%c is missing a required argument\n", optopt);
+				return EXIT_FAILURE;
+			}		
+			break;
 
 		case 'l':
 			log = 1;
+			
+			if (process_logname_arg(optarg) != 0) {
+				is_auto_log_name = 0;
+				print_usage(basename(argv[0]));
+				exit(1);
+			}
```

Comment:
Can you please think about an integration into a logrotated configuration, that can cope with the assumptions:

1. The option '-l' remains untouched
2. The logfile is **closed** and a **new logfile is opened** with a **new data-generated filename**

So no re-open/append stuff and no change with the naming.
You might also execute candump in `/tmp/candump/` to place the files there for logrotation post-processing.

I'm fine with creating two file splitting triggers about the length and the SIGHUP. But the current approach just adds too much complexity that does not need to be inside candump.